src/adapters/SwivelAdapter.sol

```diff
@@ -104,16 +105,18 @@ contract SwivelAdapter is IAdapter {
         }
 
         // get the amount of the orders
+        uint256 total;
+        uint256 fee;
         {
-            uint256 total;
             for (uint256 i = 0; i < amounts.length; ) {
                 total += amounts[i];
 
                 // extract fee
                 if (i == amounts.length - 1) {
-                    uint256 fee = total / 1000; // todo: fetch feenominator
+                    fee = total / feenominator;
                     amounts[i] = amounts[i] - fee;
                     total = total - fee;
+                    fees[underlying] += fee;
```

Comment:
Ah got it -- For some reason on my first pass i thought this was calculation of the Swivel fee rather than illuminate fee.